server/src/main/java/org/elasticsearch/index/mapper/FieldNamesFieldMapper.java

```diff
@@ -93,21 +93,28 @@ public FieldNamesFieldMapper build() {
                         "field_names_enabled_parameter", ENABLED_DEPRECATION_MESSAGE);
                 }
             }
-            FieldNamesFieldType fieldNamesFieldType = new FieldNamesFieldType(enabled.getValue().value());
-            return new FieldNamesFieldMapper(enabled.getValue(), indexVersionCreated, fieldNamesFieldType);
+            return new FieldNamesFieldMapper(enabled.getValue(), indexVersionCreated);
         }
     }
 
     public static final TypeParser PARSER = new ConfigurableTypeParser(
-        c -> new FieldNamesFieldMapper(Defaults.ENABLED, c.indexVersionCreated(), new FieldNamesFieldType(Defaults.ENABLED.value())),
+        c -> new FieldNamesFieldMapper(Defaults.ENABLED, c.indexVersionCreated()),
         c -> new Builder(c.indexVersionCreated())
     );
 
     public static final class FieldNamesFieldType extends TermBasedFieldType {
 
+        private static final FieldNamesFieldType ENABLED = new FieldNamesFieldType(true);
+
+        private static final FieldNamesFieldType DISABLED = new FieldNamesFieldType(false);
+
         private final boolean enabled;
 
-        public FieldNamesFieldType(boolean enabled) {
+        public static FieldNamesFieldType of(boolean enabled) {
```

Comment:
Maybe make this `get`, to align with `RoutingFieldMapper`? And does it need to be public?

Reply:
++ renamed, we need this exposed for some tests that's why it's `public`.